pkg/controller/installation/validation.go

```diff
@@ -336,6 +336,25 @@ func validateCustomResource(instance *operatorv1.Installation) error {
 				return fmt.Errorf("spec.calicoNetwork.containerIPForwarding is supported only for Calico CNI")
 			}
 		}
+
+		if instance.Spec.CalicoNetwork.Sysctl != nil {
+			// CNI tuning plugin
+			pluginData := instance.Spec.CalicoNetwork.Sysctl
+
+			// sysctl settings
+			allowedKeys := map[string]bool{
+				"net.ipv4.tcp_keepalive_intvl":  true,
+				"net.ipv4.tcp_keepalive_probes": true,
+				"net.ipv4.tcp_keepalive_time":   true,
+			}
+
+			for _, setting := range pluginData {
+				if !allowedKeys[setting.Key] {
+					return fmt.Errorf("key %s is not allowed in spec.calicoNetwork.sysctl", setting.Key)
+				}
+			}
```

Comment:
I'd like to see this extracted to a utility function that we can use here and in the migration code.
I'd suggest adding a function to `pkg/controller/installation/utils`, either to a new file or an existing one if you find a good place for it.

Reply:
new function created in utils to check allowed keys. However I found another bug: the migration logic were refactored in a way that finally did nothing useful.. i also fixed that, so now it is not needed to replace it with checker function.